overlord/hookstate/hookmgr.go

```diff
@@ -277,9 +288,21 @@ func snapCmd() string {
 	return filepath.Join(filepath.Dir(exe), "../../bin/snap")
 }
 
-func runHookAndWait(snapName string, revision snap.Revision, hookName, hookContext string, tomb *tomb.Tomb) ([]byte, error) {
+func killemAll(cmd *exec.Cmd) error {
+	pgid, err := syscall.Getpgid(cmd.Process.Pid)
+	if err != nil {
+		return err
+	}
+	return syscall.Kill(-pgid, 9)
+}
+
+func runHookAndWait(snapName string, revision snap.Revision, hookName, hookContext string, maxRuntime time.Duration, tomb *tomb.Tomb) ([]byte, error) {
 	command := exec.Command(snapCmd(), "run", "--hook", hookName, "-r", revision.String(), snapName)
 
+	// setup a process group for the command so that we can kill parent
```

Comment:
This is only mildly reliable but for the sake of landing this quickly I will not block on this.

I would recommend that we instead use systemd's ability to create transient units to create a service that can be reliably stopped. This can be experimented with on command line with `systemd-run`. You can run and stop a service without having to reimplement (poorly) everything required. On top of this we can provide resource control for hooks (no fork bombs) and many other useful things, I'm sure. Each transient service gets a service identifier that we can reliably stop or kill.